NDTensors/ext/NDTensorsGPUArraysCoreExt/permutedims.jl

```diff
@@ -0,0 +1,15 @@
+using NDTensors: NDTensors, array
+using GPUArraysCore: @allowscalar, AbstractGPUArray
+using NDTensors.Adapt
+function NDTensors.permutedims!(
+  Rexposed::Exposed{<:AbstractGPUArray,<:DiagTensor},
+  texposed::Exposed{<:AbstractGPUArray,<:DiagTensor},
+  perm::NTuple{N,Int},
+  f::Function=(r, t) -> t,
+) where {N}
+  R = unexpose(Rexposed)
+  t = unexpose(texposed)
+
+  array(R) .= f.(array(R), array(t))
```

Comment:
No scalar indexing is thrown but also the data does not properly get moved to R and the output is numerically incorrect. Working on that

Reply:
Using data fixed this issue